ui/src/components/admin/Notifications/PublishProgress.js

```diff
@@ -162,22 +162,17 @@ const PublishProgress = ({ renderIcon }) => {
     };
 
     useEffect(() => {
-      if (!prevPublishState) {
-        prevPublishState.current = getBulkPublishState();
-        return;
-      }
-
       if (
         refreshModelsTable &&
         (getBulkPublishState() === BulkPublishState.complete ||
           getBulkPublishState() === BulkPublishState.stopped) &&
-        prevPublishState.current === BulkPublishState.publishing
+        (prevPublishState.current === BulkPublishState.publishing || !prevPublishState.current)
```

Comment:
Also similarly to the other file, adding a null check here so that we trigger the refresh in the event that the publish finishes before this useEffect is called, ensuring the table data gets updated after it completes.